src/casync-http.c

```diff
@@ -30,6 +31,40 @@ typedef enum ProcessUntil {
         PROCESS_UNTIL_FINISHED,
 } ProcessUntil;
 
+static CURLcode robust_curl_easy_perform(CURL *curl) {
+        useconds_t sleep_time = 100000;
+        uint trial = 1;
+        uint limit = 10;
+        CURLcode c;
+
+        assert(curl);
+
+        while (trial < limit) {
+
+                c = curl_easy_perform(curl);
+
+                switch (c) {
+
+                case CURLE_COULDNT_CONNECT:
+                        /* Although this is not considered as a transient error by curl,
+                         * this error can happen momentarily while casync is retrieving
+                         * all the chunks from a remote. In this case we want to give
+                         * a break to the server and retry.
+                         */
+                        usleep(sleep_time);
+                        trial++;
+                        sleep_time *= trial;
```

Comment:
This grows too fast (it is a factorial), at step 5 it will sleep for 12 seconds, at step 6 it will sleep for 72 seconds...step 9 = 10 hours..:)
I think you should remove this assignment and use 'usleep(sleep_time * trial)' above instead of 'usleep(sleep_time)'.

Reply:
Oups, that's a bit embarrassing indeed... I fixed according to your suggestion, thanks.